### DISCUSSION TITLE
File Logging?

### DISCUSSION THREAD

**Discussion by RapidTransit** [zitadel/zitadel#5251] [Q&A] [answered]
I can't for the life of me figure out where and how to do external file logging, is there an option?

**Comment by eliobischof** [comment, marked as answer]
Hi @RapidTransit 

ZITADEL doesn't support file logging, because a twelve factor app shouldn't write logs to files. https://12factor.net/logs

Log file management should not be in each business apps responsibility. Instead, your execution environment should provide tooling for managing logs in a generic way. This includes tasks like rotating files, routing, collecting, archiving and cleanupping. For example, systemd has journald and kubernetes has fluentd and fluentbit.

If you just wan't to have the ZITADEL logs in a file, you can always redirect from stdout, of couse. But at some point, you will have to ensure that your disk doesn't run out of space.

**Reply by RapidTransit** [reply]
Ah, I'm used to the JVM where writing to stdout can be  expensive.  (I guess we are spoiled by the log4j and logback, they can automatically rotate, purge, etc)
